framework/src/base/MooseApp.C

```diff
@@ -215,6 +221,19 @@ validParams<MooseApp>()
       false,
       "Keep standard output from all processors when running in parallel");
 
+  // Options for debugging
+  params.addCommandLineParam<std::string>("start_in_debugger",
+                                          "--start-in-debugger [debugger]",
+                                          "Start the application and attach a debugger.  This will "
+                                          "launch xterm windows using the command you specify for "
+                                          "'debugger'");
+
+  params.addCommandLineParam<unsigned int>("stop_for_debugger",
+                                           "--stop-for-debugger [seconds]",
```

Comment:
It's unclear to me if this is optional or required. You have a default (so the syntax looks correct), but you don't appear to ever use the default in the code below.

Reply:
It's definitely not required.  It's just a default _value_ so you don't have to specify a value.